### PR TITLE
[WIP] Tests for autonomy package

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR adds tests for autonomy package

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: autonomy/cli/hash.py [hunk 1]
```diff
@@ -174,7 +174,7 @@ def generate_all(
 @hash_group.command(name="one")
 @click.argument("path", type=click.Path(exists=True, file_okay=True, dir_okay=True))
 @click.option("--no-wrap", is_flag=True)
-def hash_file(path: str, no_wrap: bool) -> None:
+def hash_file(path: str, no_wrap: bool) -> None:  # pragma: nocover
```
Comment (angrybayblade) [inline autonomy/cli/hash.py:177]: TODO: remove no cover marker and add tests

File: autonomy/analyse/abci/logs.py [hunk 2]
```diff
@@ -71,13 +71,13 @@ def parse_file(  # pylint: disable=too-many-locals
             m = re.match(EVENT_REGEX, line)
             if m:
                 agent_id = m.groupdict()["agent_id"]
-                if agent_id not in periods.keys():
+                if agent_id not in periods.keys():  # pragma: nocover
```
Comment (DavidMinarsch) [inline autonomy/analyse/abci/logs.py:74]: It would be good to have tests for these cases too?

File: autonomy/analyse/abci/handlers.py [hunk 3]
```diff
@@ -42,7 +42,7 @@ def check_handlers(
     try:
         module = importlib.import_module(module_name)
         module_attributes = dir(module)
-    except ModuleNotFoundError as exc:
+    except ModuleNotFoundError as exc:  # pragma: nocover
```
Comment (DavidMinarsch) [inline autonomy/analyse/abci/handlers.py:45]: Ideally, we have tests for these cases too.

File: autonomy/cli/deploy.py [hunk 4]
```diff
@@ -125,12 +125,12 @@ def build_deployment(  # pylint: disable=too-many-arguments
     build_dir = Path(output_dir, "abci_build")
 
     if not packages_dir.is_dir():
-        raise click.ClickException(
+        raise click.ClickException(  # pragma: nocover
```
Comment (DavidMinarsch) [inline autonomy/cli/deploy.py:128]: See above, we should also test our error handling!

File: autonomy/cli/fetch.py [hunk 5]
```diff
@@ -98,7 +98,7 @@ def fetch_service(ctx: Context, public_id: PublicId) -> None:
     if ctx.registry_type == REGISTRY_REMOTE:
         if get_default_remote_registry() == REMOTE_IPFS:
             fetch_service_ipfs(public_id)
-        else:
+        else:  # pragma: nocover
```
Comment (DavidMinarsch) [inline autonomy/cli/fetch.py:101]: Let's test these ignored branches too

File: autonomy/cli/hash.py [hunk 6]
```diff
@@ -125,13 +125,13 @@ def update_hashes(  # pylint: disable=too-many-locals
                 public_id_to_hash_mappings[package_id] = package_hash
 
                 if vendor is not None and package_id.author != vendor:
-                    continue
+                    continue  # pragma: nocover
```
Comment (DavidMinarsch) [inline autonomy/cli/hash.py:128]: Please include in tests

File: autonomy/cli/hash.py [hunk 7]
```diff
@@ -160,7 +160,7 @@ def generate_all(
 ) -> None:
     """Generate IPFS hashes."""
     packages_dir = Path(packages_dir).absolute()
-    if check:
+    if check:  # pragma: nocover
```
Comment (DavidMinarsch) [inline autonomy/cli/hash.py:163]: Test too

**Review by DavidMinarsch — CHANGES_REQUESTED** [review]
Good progress but I would like to see less branches ignored. Testing these negative cases is also helpful!

File: autonomy/analyse/abci/app_spec.py [hunk 8]
```diff
@@ -35,7 +35,7 @@ class DFASpecificationError(Exception):
     """Simple class to raise errors when parsing a DFA."""
 
 
-class DFA:
+class DFA:  # pragma: nocover
```
Comment (angrybayblade) [inline autonomy/analyse/abci/app_spec.py:38]: I think Jose would be more suitable to write tests for this class.

File: autonomy/analyse/abci/logs.py [hunk 9]
```diff
@@ -109,5 +109,5 @@ def parse_file(  # pylint: disable=too-many-locals
             )
 
     print("\nERRORS:")
-    for error in errors:
+    for error in errors:  # pragma: nocover
```
Comment (angrybayblade) [inline autonomy/analyse/abci/logs.py:112]: Same goes with this function, I think David V would be more suitable to write tests for this method.

File: autonomy/cli/fetch.py [hunk 10]
```diff
@@ -84,7 +84,7 @@ def fetch(
     ctx.registry_type = registry
 
     try:
-        if package_type == AGENT:
+        if package_type == AGENT:  # pragma: nocover
             do_fetch(ctx, public_id, alias)
```
Comment (angrybayblade) [inline autonomy/cli/fetch.py:88]: This is already being tested on open-aea.

File: autonomy/cli/publish.py [hunk 11]
```diff
@@ -66,11 +66,13 @@ def publish(
     try:
         if Path(click_context.obj.cwd, DEFAULT_SERVICE_CONFIG_FILE).exists():
             publish_service_package(click_context, registry)
-        elif Path(click_context.obj.cwd, DEFAULT_AEA_CONFIG_FILE).exists():
+        elif Path(
+            click_context.obj.cwd, DEFAULT_AEA_CONFIG_FILE
+        ).exists():  # pragma: nocover
             publish_agent_package(click_context, registry, push_missing)
```
Comment (angrybayblade) [inline autonomy/cli/publish.py:72]: Same as above

File: autonomy/cli/replay.py [hunk 12]
```diff
@@ -104,7 +105,7 @@ def run_tendermint(build_dir: Path) -> None:
         tendermint_network.stop()
 
 
-def load_docker_config(file_path: Path) -> Dict[str, Any]:
+def load_docker_config(file_path: Path) -> Dict[str, Any]:  # pragma: nocover
     """Load docker config."""
     with open(str(file_path), "r", encoding="utf-8") as fp:
         docker_compose_config = yaml.safe_load(fp)
```
Comment (angrybayblade) [inline autonomy/cli/replay.py:111]: `load_docker_config` is patched during tests and this logic is only a wrapper around `yaml.safe_load` so I don't think we need to test this.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**